spline/bezier.go

```diff
@@ -0,0 +1,172 @@
+// TODO:
+// 1: type Bezier (points and control points) (done)
+// 2: open to factors 5th-order Bezier (not needed because 5 is done)
+// 3: derivatives (done)
+// 4: get/set optimization (done)
+
+// 5: general n-th order open to factors (done)
+// 6: smoothing with other Bezier (not done yet)
+
+package spline
+
+import (
+	"math"
+)
+
+//this includes edge points and control points
+//when smoothing with other bezier we might need to seperate edge from control points
+type BezierPoint struct {
+	X float64
+	Y float64
+}
+
+type Bezier struct {
+	points []BezierPoint
```

Comment:
The seperation between normal points and control points needs to be written now - the function "NewBezierCurve" should accept only "edge points" and degree (will determine the amount of control points) and then it will generate the bezier control points by itself.